specs/number-field.md

```diff
@@ -0,0 +1,123 @@
+- [Office Fabric](https://developer.microsoft.com/en-us/fabric#/controls/web/spinbutton)
+- [Duet](https://www.duetds.com/components/number-input/)
+
+---
+
+### API
+
+Extends Form associated component
+
+*Component Name*
+- `fast-number-field`
+
+*Attrs* - Adapted from [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number)
+- `autofocus` - automatically focuses the control
+- `readonly` - the number field should be submitted with the form but should not be editable.
+- `placeholder` - an exemplar value to display in the input field whenever it is empty
+- `required` - boolean value that sets the field as required
+- `value` - string value of the number field, can be an empty string
+- `disabled` - disables the control
+- `list` - the id of the `<datalist>` element that contains the optional pre-defined autocomplete options
+- `max`	- the maximum value to accept for this input
+- `min` -	the minimum number to accept for this input
+- `maxlength`	- the maximum number of characters the input should accept
+- `minlength` -	the minimum number of characters long the input can be and still be considered valid
+- `name` - the name of the control
+- `size` - a number indicating how many characters wide the input field should be
+- `appearance` - enum
+  - outline (default)
+  - filled
+
+*Events*
+- `change: CustomEvent`
+  - no custom data
+  - bubbles
+
+### Anatomy and Appearance
+
+```HTML
+<!-- shadow root -->
+<label part="label"><slot></slot></label>
+<div part="root">
+  <slot name="start" part="start"></slot>
+  <input type="number" part="control" />
+  <slot name="end" part="end"></slot>
+</div>
+<!-- end shadow root -->
+```
+
+
+*Screenshots below are of the basic appearance of the component and are not exhaustive.*
+
+**The visual treatment of this control should visually match the number field component**
+
+| State | Image |
+| ----- | ----- |
+| default | ![](../text-field/images/text-field.png) |
+| focus | ![](../text-field/images/text-field-focus.png)
+| filled | ![](../text-field/images/text-field-filled.png)
+
+*Slot Names*
+- default - the label content
+- start - often times a glyph, icon, or button precedes input
+- end - often times a glyph, icon, or button follows the input
+
+*Host Classes*
+- disabled
+- required
+- readonly
+
+*CSS Parts*
+- label
+- root
+- start
+- end
+- control
+
+---
+
+## Implementation
+
+### States
+
+**disabled**: `true` or `false` - when disabled, the value will not be changeable through user interaction. It should also not expose it's value to a form submission.
+
+**readonly**: `true` or `false` - when readonly, the value will not be changeable through user interaction. The value will still be exposed to forms on submission.
+
+### Accessibility
+
+The host element will be the focusable element and should handle providing the role of textbox to assistive technology (AT) as well as reporting any state based aria attributes (disabled, readonly, etc) to AT.
```

Comment:
Are there any keyboard interactions for increment/decrement of the number?

Reply:
Sure, I can articulate that.